core/src/maybe_uninit.rs

```diff
@@ -0,0 +1,41 @@
+//! This module contains a re-export or vendored version of `core::mem::MaybeUninit` depending
+//! on which Rust version it's compiled for.
+//!
+//! Remove this module and use `core::mem::MaybeUninit` directly when dropping support for <1.36
+#![allow(dead_code)]
+
+#[cfg(has_maybe_uninit)]
+pub use core::mem::MaybeUninit;
+
+#[cfg(not(has_maybe_uninit))]
+use core::mem::ManuallyDrop;
+
+/// Copied from `core::mem::MaybeUninit` to support Rust older than 1.36
+#[cfg(not(has_maybe_uninit))]
+pub union MaybeUninit<T: Copy> {
+    uninit: (),
+    value: ManuallyDrop<T>,
+}
+
+#[cfg(not(has_maybe_uninit))]
+impl<T: Copy> MaybeUninit<T> {
+    #[inline(always)]
+    pub fn uninit() -> MaybeUninit<T> {
+        MaybeUninit { uninit: () }
+    }
+
+    #[inline(always)]
```

Comment:
Is `#[inline(always)]` really necessary here? Would plain `#[inline]` be sufficient?

Reply:
I can change it. I just copied the implementation and annotations from `libcore`

Reply:
I have not looked at any generated code to see whether it makes a difference.

Reply:
I would prefer changing it to a simple `#[inline]`.

Reply:
Ok. I personally don't really see why we would want to write the copied version differently than the original. But I can push that! Do you mean for this method in particular or everything on our `MaybeUninit`?

Reply:
For all of the methods.

Reply:
Done